### DISCUSSION TITLE
Session layer edits for the pull/push workflow question

### DISCUSSION THREAD

**Discussion by voidreamer** [Autodesk/maya-usd#3761] [Q&A]
Hi, where can I find the callback or procedure that makes maya store the session layer edit when doing a pull/push maya reference in a stage ?
or what's the best workflow to use pull/push without embedding in the maya file ?
I'm making a tool where I want to store the maya file without embedded edits, when I use the option of `Save Maya scene and USD layers` I still get the session layer with the edits for the maya reference, but when I use save maya scene only I lose them when reopening as I show here:

`Save Maya scene only.`
![02](https://github.com/Autodesk/maya-usd/assets/9424247/907325f6-fe77-4db7-80fb-92ab04fe62bf)

Maya and Usd layers (preserves the session layer edit)
![01](https://github.com/Autodesk/maya-usd/assets/9424247/c81f3a6c-bb32-4152-b250-fb8b02b7455b)

Thanks

**Comment by voidreamer** [comment]
I ended up serializing the session layer and restoring it on opening the file.
Not sure if that's the suggested approach.

is there a way to get or generate  the dictionary pull edit this workflow needs by myself ?
and how to get the parent `|__mayaUsd__|MayaReference1Parent|MayaReference1` ?

**Comment by wallworm** [comment]
Sorry this went under the radar. I'll bring it up with the team.

**Comment by wallworm** [comment]
After talking to the devs, they said that you are already doing the best approach for getting/setting the session. I'll look more into your other question.

**Reply by voidreamer** [reply]
thank you !  I really appreciate it, I'm still working on it : )
